source/parser.cpp

```diff
@@ -0,0 +1,98 @@
+#include "parser.hpp"
+#include <stdexcept>
+#include <string>
+#include <yaml-cpp/yaml.h>
+
+namespace sim {
+	void YamlParser::parseConfig(const std::string &filename, Simulator &simulator) {
+		const YAML::Node config = YAML::LoadFile(filename);
+		processHosts(config, simulator);
+		processSwitches(config, simulator);
+		processLinks(config, simulator);
+		simulator.recalculate_paths();
```

Comment:
```suggestion
		devices_map.clear();
```
Simulator should be the only one owner of devices so parser must delete all `shared_ptr`s

Reply:
Added deletion of `shared_ptr` s

Reply:
Done